cirq/protocols/control.py

```diff
@@ -27,7 +27,7 @@
 TDefault = TypeVar('TDefault')
 
 
-def control(controllee: Union['cirq.Gate', op_tree.OP_TREE],
+def control(controllee: op_tree.OP_TREE,
```

Comment:
I think we should delete this entire file (`cirq/protocols/control.py`). With gate removed, this protocol is effectively dominated by the `controlled_by` method on operation. Repeating across lists isn't enough of an "add on" to justify keeping it.

Now is also the perfect time to do it, since the last release didn't have it; it will avoid a breaking change.

Reply:
I was planning to add support for gates in the subsequent PR that addressed the TODO in this file. It would throw error if the control qubits argument is specified when the operand is a Gate. But it would still be able to use the Gate.controlled() method with control_values and control_qid_shape. Does that sound like something worth adding or should I continue with deleting the file?

Reply:
I would continue with deleting it. However, that can be a separate PR.